### ISSUE TITLE
Android Basics: Viewmodel and State in Compose

### ISSUE THREAD

**Issue by gloglo1980** [google-developer-training/basic-android-kotlin-compose-training-unscramble#210]
This is the first lesson I found difficult to follow. I feel like there was a lot of jumping around and doing things and I didn't really know why. It could have been organised differently - for example, I don't feel like we (learners) gain value from adding things one by one in the view model and the UI, it might have been better to have started with a plan: we're going to have a guessed word, a score and a word count that will need to behave like x,y, and z, and then executed once in each place and had more focus on the form these things take. For example there was no talk about "_uiState" - is it a convention to define this kind of variable with an underscore in front? The writer knew upfront what this variable was for but we (learners) were just along for the ride and had to infer with no experience.

**Comment by alarimer** [comment]
> This is the first lesson I found difficult to follow. I feel like there was a lot of jumping around and doing things and I didn't really know why. It could have been organised differently - for example, I don't feel like we (learners) gain value from adding things one by one in the view model and the UI, it might have been better to have started with a plan: we're going to have a guessed word, a score and a word count that will need to behave like x,y, and z, and then executed once in each place and had more focus on the form these things take. For example there was no talk about "_uiState" - is it a convention to define this kind of variable with an underscore in front? The writer knew upfront what this variable was for but we (learners) were just along for the ride and had to infer with no experience.

Agreed.  This is the worst lesson that I have encountered this far.  Being a seasoned software engineer didn't help much.  There are several Forks and Issues with actionable feedback, yet there seems to be little action. :`(
